webbpsf/optics.py

```diff
@@ -1111,8 +1117,42 @@ def __init__(self, instrument, include_oversize=False, **kwargs):
 
             # Cubic interpolation of of non-uniform 2D grid
             cf = griddata((v2, v3), zvals, (v2_tel, v3_tel), method='cubic').tolist()
+            # Want to perform extrapolation if outside of bounds
             if np.isnan(cf):
-                cf = self.row[zkey]
+                #cf = self.row[zkey]
+
```

Comment:
Perhaps add here this explanatory text, adapted from your comment on #246:

```
    # To extrapolate outside the measured field points, we proceed in two steps. 
    # This first creates a fine-meshed cubic fit over the known field points, 
    # fixes any NaN's using RegularGridInterpolator, then again uses 
    # RegularGridInterpolator on the fixed data to extrapolate the requested field point.

    # In principle, the first call of RegularGridInterpolator can be used to extrapolate 
    # the requested field point to eliminate the intermediate step, but this method
    # enables use of all the real data rather than the trimmed data set.
```